app/ImportExportHelpers/ScenarioImportExportHelper.php

```diff
@@ -212,11 +212,13 @@ public static function patchScenario(Scenario $persistedScenario, Scenario $scen
             ConversationDataClient::updateScenario($scenarioPatch);
         }
 
-        foreach ($scenarioWithPathsSubstituted->getConversations() as $cIdx => $conversationWithPathsSubstituted) {
+        foreach ($scenarioWithPathsSubstituted->getConversations() as $conversationWithPathsSubstituted) {
             /** @var Conversation $conversationWithPathsSubstituted */
 
             /** @var Conversation $persistedConversation */
-            $persistedConversation = $persistedScenario->getConversations()[$cIdx];
+            $persistedConversation = $persistedScenario->getConversations()->filter(
+                fn (Conversation $conversation) => $conversation->getOdId() === $conversationWithPathsSubstituted->getOdId()
+            )->first();
```

Comment:
A minor point, but each `ODObjectCollection` subclass has a `getObjectsWithId` method which does this.

```suggestion
            $persistedConversation = $persistedScenario->getConversations()->getObjectsWithId($conversationWithPathsSubstituted->getOdId())->first();
```